src-docs/src/views/combo_box/combo_box_example.js

```diff
@@ -567,5 +573,28 @@ export const ComboBoxExample = {
       snippet: startingWithSnippet,
       demo: <StartingWith />,
     },
+    {
+      title: 'Duplicate labels',
+      source: [
+        {
+          type: GuideSectionTypes.JS,
+          code: duplicateOptionsSource,
+        },
+        {
+          type: GuideSectionTypes.HTML,
+          code: duplicateOptionsHtml,
+        },
+      ],
+      text: (
+        <p>
+          If you want to use options with a duplicate label (which in general is
+          not recommended since the user has no way to distinguish those
+          options), you need to set a unique <EuiCode language="js">id</EuiCode>{' '}
+          for each option.
```

Comment:
I would start with the caveat so it's more likely to be read.

```suggestion
          In general, it is not recommended to use duplicate labels on the options
          because the user has no way to distinguish between them. If you need 
          duplicate labels, you will need to add a unique{' '}
          <EuiCode language="js">id</EuiCode> for each option.
```